File: backend/internal/model/campus.go
package model

import "time"

type CampusCalendar struct {
	ID          int64     `json:"id"`
	Title       string    `json:"title"`
	EventDate   string    `json:"event_date"`
	EventType   string    `json:"event_type"`
	Description string    `json:"description"`
	CreatedAt   time.Time `json:"created_at"`
}

type CampusDirectory struct {
	ID         int64     `json:"id"`
	Name       string    `json:"name"`
	Department string    `json:"department"`
	Position   string    `json:"position"`
	Phone      string    `json:"phone"`
	Email      string    `json:"email"`
	Office     string    `json:"office"`
	Category   string    `json:"category"`
	CreatedAt  time.Time `json:"created_at"`
}

type EmptyClassroom struct {
	ID        int64     `json:"id"`
	Building  string    `json:"building"`
	Room      string    `json:"room"`
	Capacity  int       `json:"capacity"`
	Weekday   int       `json:"weekday"`
	StartTime string    `json:"start_time"`
	EndTime   string    `json:"end_time"`
	Semester  string    `json:"semester"`
	CreatedAt time.Time `json:"created_at"`
}

File: frontend/src/api/campus.ts
import { request } from "./request"

export interface CalendarEvent {
  id: number
  title: string
  event_date: string
  event_type: string
}

export interface DirectoryEntry {
  id: number
  name: string
  department: string
  position: string
  phone: string
  email: string
  office: string
  category: string
}

export interface EmptyRoom {
  id: number
  building: string
  room: string
  capacity: number
  weekday: number
  start_time: string
  end_time: string
}

export function listCalendar() {
  return request<CalendarEvent[]>({ url: "/campus/calendar", method: "GET" })
}

export function listDirectory(department?: string) {
  return request<DirectoryEntry[]>({ url: "/campus/directory", method: "GET", data: department ? { department } : {} })
}

export function listBuildings() {
  return request<string[]>({ url: "/campus/rooms/buildings", method: "GET" })
}

export function listEmptyRooms(building?: string, weekday?: number) {
  return request<EmptyRoom[]>({ url: "/campus/rooms", method: "GET", data: { building, weekday } })
}
